### DISCUSSION TITLE
Guidance Required: WKT or vector polygon to bitmap/raster layer

### DISCUSSION THREAD

**Discussion by robalexclark** [SharpMap/SharpMap#203] [Q&A]
Can someone please give me some pointers to achieve the following:
I have a WKT string (set of polygons) that I would like to turn into a bitmap/ raster layer that can be consumed in openlayers. Can Sharpmap achieve this? At the moment the WKT is downloaded to openlayers and processed clientside, which can take some time if there are lots of polygons.

For example see the following test page which requires the client to download lots of polygons in json format to openlayers before the map is usable:

https://floodassist.co.uk/flood-warnings/test-map

Just need some general direction to understand how to do this.

**Comment by FObermaier** [comment]
SharpMap can turn your vector data into bitmap images.
You need to set up a `SharpMap.Map` and add `SharpMap.Layers.ILayer`s to it. There is e.g. `SharpMap.Layers.VectorLayer` that can cope with vector data. Choose a `SharpMap.Data.Providers.IProvider` that does data retrival for you.

Have a look at [this set of samples](https://github.com/SharpMap/SharpMap/tree/develop/Examples/WinFormSamples/Samples) to get you started.
